static/app/components/modals/inviteMembersModal/index.spec.tsx

```diff
@@ -11,56 +11,100 @@ import {makeCloseButton} from 'sentry/components/globalModal/components';
 
 jest.mock('sentry/utils/useOrganization');
 
 describe('InviteMembersModal', function () {
-  const team = TeamFixture();
-  const org = OrganizationFixture({access: ['member:write'], teams: [team]});
-  TeamStore.loadInitialData([team]);
-
   const styledWrapper = styled(c => c.children);
-  const modalProps: ComponentProps<typeof InviteMembersModal> = {
+
+  type MockApiResponseFn = (
+    client: typeof MockApiClient,
+    orgSlug: string,
+    roles?: object[]
+  ) => jest.Mock;
+  const defaultMockOrganizationRoles: MockApiResponseFn = (client, orgSlug, roles) => {
+    return client.addMockResponse({
+      url: `/organizations/${orgSlug}/members/me/`,
+      method: 'GET',
+      body: {roles},
+    });
+  };
+
+  const defaultMockPostOrganizationMember: MockApiResponseFn = (client, orgSlug, _) => {
+    return client.addMockResponse({
+      url: `/organizations/${orgSlug}/members/`,
+      method: 'POST',
+    });
+  };
+
+  const defaultMockModalProps = {
     Body: styledWrapper(),
     Header: p => <span>{p.children}</span>,
     Footer: styledWrapper(),
     closeModal: () => {},
     CloseButton: makeCloseButton(() => {}),
   };
 
-  const noWriteOrg = OrganizationFixture({
-    access: [],
-  });
+  const setupView = ({
+    orgTeams = [TeamFixture()],
+    orgAccess = ['member:write'],
+    roles = [
+      {
+        id: 'admin',
+        name: 'Admin',
+        desc: 'This is the admin role',
+        allowed: true,
+      },
+      {
+        id: 'member',
+        name: 'Member',
+        desc: 'This is the member role',
+        allowed: true,
+      },
+    ],
+    modalProps = defaultMockModalProps,
+    mockApiResponses = [defaultMockOrganizationRoles],
+  }: {
+    mockApiResponses?: MockApiResponseFn[];
+    modalProps?: ComponentProps<typeof InviteMembersModal>;
+    orgAccess?: Scope[];
+    orgTeams?: DetailedTeam[];
+    roles?: object[];
+  } = {}) => {
+    const org = OrganizationFixture({access: orgAccess, teams: orgTeams});
+    TeamStore.reset();
+    TeamStore.loadInitialData(orgTeams);
 
-  const roles = [
-    {
-      id: 'admin',
-      name: 'Admin',
-      desc: 'This is the admin role',
-      allowed: true,
-    },
-    {
-      id: 'member',
-      name: 'Member',
-      desc: 'This is the member role',
-      allowed: true,
-    },
-  ];
-
-  beforeEach(function () {
     MockApiClient.clearMockResponses();
-    MockApiClient.addMockResponse({
-      url: `/organizations/${org.slug}/members/me/`,
-      method: 'GET',
-      body: {roles},
+    const mocks: jest.Mock[] = [];
+    mockApiResponses.forEach(mockApiResponse => {
+      mocks.push(mockApiResponse(MockApiClient, org.slug, roles));
     });
-  });
-
-  it('renders', async function () {
     jest.mocked(useOrganization).mockReturnValue(org);
-    render(<InviteMembersModal {...modalProps} />);
 
+    return {...render(<InviteMembersModal {...modalProps} />), mocks};
+  };
+
+  const setupMemberInviteState = async () => {
```

Comment:
is there anyway we can do this by manipulating the state instead of user events? If we can, it adds a bit of stability to the test.

Reply:
I'll have to dig a little more and understand what to pass, but I think we can do this. Let me do this in a follow up so the changes are easier to review?